### PR TITLE
Building audio/annotation paths in core

### LINKED ISSUE #442: core.track_path(key)
The loaders have a very similar repeated line all over the code:
`self.beats_path = os.path.join(self._data_home, self._track_paths["beats"][0])`
`self.audio_path = os.path.join(self._data_home, self._track_paths["audio"][0])`
`...`

We should generalize this to a function in core

### PR DESCRIPTION
**Pull request by genisplaja**
Fixes #442

It is basically a `get_path` function defined in the Track class. It uses the `none_path_join` function as well. At the moment, I just updated the OTMM Makam Dataset. Please, let me know what you think and I will update the rest:)

### CODE REVIEW

**Review by magdalenafuentes — COMMENTED** [review]
@genisplaja this is looking great! 👍

**Comment by genisplaja** [comment]
I think it's done! Feel free to review it:)

**Comment by nkundiushuti** [comment]
in my case the audio_paths is a list of paths. this is because each track has a variable number of audio mics.

**Comment by genisplaja** [comment]
> in my case the audio_paths is a list of paths. this is because each track has a variable number of audio mics.

@nkundiushuti I see! Maybe a solution would be to add a conditional in the function `get_path` to check whether audio_path is a list or not, and if it is, iteratively compute the paths and return them in a form of a list as well.

File: mirdata/core.py [hunk 1]
```diff
@@ -380,6 +380,9 @@ def __repr__(self):
     def to_jams(self):
         raise NotImplementedError
 
+    def get_path(self, key):
```
Comment (rabitt) [inline mirdata/core.py:383]: perfect! For the final PR, dont forget to add a docstring for this function.

File: mirdata/datasets/giantsteps_key.py [hunk 2]
```diff
@@ -112,13 +112,13 @@ def __init__(
...
         )
 
-        self.audio_path = os.path.join(self._data_home, self._track_paths["audio"][0])
-        self.keys_path = os.path.join(self._data_home, self._track_paths["key"][0])
+        self.keys_path = self.get_path("key")
         self.metadata_path = (
-            os.path.join(self._data_home, self._track_paths["meta"][0])
-            if self._track_paths["meta"][0] is not None
-            else None
+            self.get_path("meta") if self._track_paths["meta"][0] is not None else None
```
Comment (rabitt) [inline mirdata/datasets/giantsteps_key.py:117]: shouldn't the `none_path_join` remove the need for the if/else? i.e. i think you can directly call `self.metadata_path = self.get_path("meta")` and the result will be the same

**Review by rabitt — COMMENTED** [review]
This is great! Two small requests - 
1. Update the contributing example.py for the docs
2. add a small test to `tests/test_core.py` (maybe a new test called test_track) that runs a few minimal tests on a toy example for the new function.

**Comment by rabitt** [comment]
@genisplaja I'm making a small commit to add a test. Then this is ready to go!

**Comment by rabitt** [comment]
@genisplaja feel free to merge this!

**Review by rabitt — APPROVED** [review]

**Comment by nkundiushuti** [comment]
@genisplaja could you merge this so I can update my branch with the changes in this branch?

**Comment by genisplaja** [comment]
Thanks @rabitt for the test, and sorry for the delay! I'm merging right now.

**Pull request merged**